hyperspy/signal.py

```diff
@@ -4005,38 +4011,68 @@ def is_rgb(self):
     def is_rgbx(self):
         return rgb_tools.is_rgbx(self.data)
 
-    def add_marker(self, marker, plot_on_signal=True, plot_marker=True):
+    def add_marker(
+            self, marker, plot_on_signal=True, plot_marker=True,
+            permanent=False):
         """
         Add a marker to the signal or navigator plot.
 
         Plot the signal, if not yet plotted
 
         Parameters
         ----------
-        marker: `hyperspy.drawing._markers`
-            the marker to add. see `plot.markers`
-        plot_on_signal: bool
+        marker : `hyperspy.drawing._markers`
+            the marker to add. See `plot.markers`
+        plot_on_signal : bool
             If True, add the marker to the signal
             If False, add the marker to the navigator
-        plot_marker: bool
+        plot_marker : bool
             if True, plot the marker
+        permanent : bool, default False
+            If False, the marker will only appear in the current
+            plot. If True, the marker will be added to the
+            markers list, and be plotted with plot(plot_markers=True).
 
         Examples
         -------
         >>> import scipy.misc
         >>> im = hs.signals.Signal2D(scipy.misc.ascent())
-        >>> m = hs.plot.markers.rectangle(x1=150, y1=100, x2=400,
+        >>> m = hs.markers.rectangle(x1=150, y1=100, x2=400,
         >>>                                  y2=400, color='red')
         >>> im.add_marker(m)
 
+        Add permanent marker
+        >>> s = hs.signals.Signal2D(np.random.random(100, 100))
+        >>> marker = hs.markers.point(50, 60)
+        >>> s.add_marker(marker, permanent=True, plot_marker=False)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
+
+        Add permanent marker which changes with navigation position
+        >>> s = hs.signals.Signal2D(np.random.randint(10, size=(3, 100, 100)))
+        >>> marker = hs.markers.point((10, 30, 50), (30, 50, 60), color='red')
+        >>> s.add_marker(marker, permanent=True, plot_marker=False)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
         """
-        if self._plot is None:
-            self.plot()
-        if plot_on_signal:
-            self._plot.signal_plot.add_marker(marker)
-        else:
-            self._plot.navigator_plot.add_marker(marker)
+        marker._plot_on_signal = plot_on_signal
```

Comment:
It would be nice to be able to plot them in different signals, but I don't think I want to implement that in this pull request. I added a check to see if the markers are already added to another signal, and will throw an `Exception` if they are. Which exception would be the best to throw here?